ddtrace/_trace/tracer.py

```diff
@@ -443,42 +441,127 @@ def configure(
+        :param bool partial_flush_enabled: This parameter is deprecated and will be removed.
+        :param bool partial_flush_min_spans: This parameter is deprecated and will be removed.
+        :param str api_version: This parameter is deprecated and will be removed.
+        :param bool compute_stats_enabled: This parameter is deprecated and will be removed.
+        :param bool appsec_enabled: This parameter is deprecated and will be removed.
+        :param bool iast_enabled: This parameter is deprecated and will be removed.
+        :param bool appsec_standalone_enabled: This parameter is deprecated and will be removed.
+        :param List[TraceProcessor] trace_processors: This parameter sets TraceProcessor (ex: TraceFilters).
+           Trace processors are used to modify and filter traces based on certain criteria.
         """
+        if settings is not None:
+            deprecate(
+                "Passing settings to ``tracer.configure(...)`` is deprecated",
+                message="Please use the trace_proccessor parameter instead of settings['FILTERS'].",
+                version="3.0.0",
+                category=DDTraceDeprecationWarning,
+            )
+            if settings.get("FILTERS") is not None:
+                trace_processors += settings.get("FILTERS")
+
+        return self._configure(
+            enabled,
+            hostname,
+            port,
+            uds_path,
+            https,
+            sampler,
+            context_provider,
+            wrap_executor,
+            priority_sampling,
+            trace_processors,
+            dogstatsd_url,
+            writer,
+            partial_flush_enabled,
+            partial_flush_min_spans,
+            api_version,
+            compute_stats_enabled,
+            appsec_enabled,
+            iast_enabled,
+            appsec_standalone_enabled,
+            True,
+        )
+
+    def _configure(
+        self,
+        enabled: Optional[bool] = None,
+        hostname: Optional[str] = None,
+        port: Optional[int] = None,
+        uds_path: Optional[str] = None,
+        https: Optional[bool] = None,
+        sampler: Optional[BaseSampler] = None,
+        context_provider: Optional[DefaultContextProvider] = None,
+        wrap_executor: Optional[Callable] = None,
+        priority_sampling: Optional[bool] = None,
+        trace_processors: Optional[List[TraceProcessor]] = None,
+        dogstatsd_url: Optional[str] = None,
+        writer: Optional[TraceWriter] = None,
+        partial_flush_enabled: Optional[bool] = None,
+        partial_flush_min_spans: Optional[int] = None,
+        api_version: Optional[str] = None,
+        compute_stats_enabled: Optional[bool] = None,
+        appsec_enabled: Optional[bool] = None,
+        iast_enabled: Optional[bool] = None,
+        appsec_standalone_enabled: Optional[bool] = None,
+        log_deprecations: bool = False,
```

Comment:
What's the purpose of setting this flag? Don't we want to log deprecations always?

Reply:
This design allows us to achieve the following:
- define all of the configuration logic in one function `tracer._configure(...)`
- Log deprecation warnings every time the public `tracer.configure(...)` is used with unsupported parameters
- Avoid logging deprecations when the private `tracer._configure(...)` is used (by default, obviously someone can override this by calling `tracer._configure(..., log_deprecations=True)`
- Require minimal updates to tests and internal logic

Reply:
So am I right in understanding that when used internally these options are not deprecated and the logs are only meant for when public users attempt to use these options via `configure()`?

Reply:
Yup, the deprecations are only for public users. The public configure function will have minimal functionality and the private one will remain unchanged (for now).